python/gstaichi/lang/field.py

```diff
@@ -257,6 +257,9 @@ class ScalarField(Field):
     def __init__(self, var):
         super().__init__([var])
 
+    def to_dlpack(self):
+        return impl.get_runtime().prog.field_to_dlpack(self._snode.ptr, 0, 0, 0)
```

Comment:
Could you move this on `pybind11` side ? This would improve performance. Otherwise, I would suggest to instantiate the dlpack object once and for all at init, and return it directly in this method.